### PR TITLE
SDIT-1623 Add ability to find an IWPDocument by id

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+    ],
+  )
+  fun getDocument(
+    @Schema(description = "The document id") @PathVariable id: Long,
+  ): ByteArray? = documentService.getDocumentById(id)
+}
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:63]: still to add - get all document ids/summaries by matching template name and bookingId

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.config.ErrorResponse
+
+@RestController
+@Validated
+@RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+class DocumentResource(private val documentService: DocumentService) {
+  @PreAuthorize("hasRole('ROLE_NOMIS_DOCUMENTS')")
+  @GetMapping("/documents/{id}", produces = ["application/x-zip-compressed"])
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:21]: a weird content type but I guess there is a reason for that?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt (hunk 1 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:63]: any reason why this returns an optional ByteArray?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:21]: wrong type

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:63]: no - it should return a ByteArray - I changed the call at the last minute to throw an exception instead

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:21]: will fix these in next PR

**Pull request merged**